util/types/json/functions.go

```diff
@@ -165,3 +165,125 @@ func extract(j JSON, pathExpr PathExpression) (ret []JSON) {
 	}
 	return
 }
+
+// autoWrapAsArray wraps input JSON into an array if needed.
+func autoWrapAsArray(j JSON, hintLength int) JSON {
+	jnew := CreateJSON(nil)
+	jnew.typeCode = typeCodeArray
+	jnew.array = make([]JSON, 0, hintLength)
+	jnew.array = append(jnew.array, j)
+	return jnew
+}
+
+// Merge merges suffixes into j according the following rules:
+// 1) adjacent arrays are merged to a single array;
+// 2) adjacent object are merged to a single object;
+// 3) a scalar value is autowrapped as an array before merge;
+// 4) an adjacent array and object are merged by autowrapping the object as an array.
+func (j JSON) Merge(suffixes []JSON) JSON {
```

Comment:
swap a and b is wrong because `json_merge('1', '2') != json_merge('2', '1')`.
and, I didn't use function `mergeTwo`, because it's not easy to pass argument `hintLength` to `autoWrapAsArray` in `mergeTwo`.